src/type/classes/StringType.ts

```diff
@@ -69,6 +91,19 @@ export class StringType extends AbstractType<schema.StringSchema> {
         ctx.js(/* js */ `if(${r}.length > ${max}) return ${err};`);
       }
     }
+    
+    // Handle format validation
+    if (format) {
+      const formatErr = ctx.err(ValidationError.STR, path);
+      const validateFn = ctx.codegen.linkDependency(validateStringFormat);
```

Comment:
Link `isAscii` and `isUtf8` dependencies directly and generate the code calling the right dependency, you know (according to schema) the string format at codegen time.